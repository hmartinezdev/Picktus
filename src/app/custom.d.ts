interface SvgrComponent extends React.StatelessComponent<React.SVGAttributes<SVGElement>> {}

declare module '*.svg' {
  const value: SvgrComponent;
  export default value;
}

interface PicktusUser {
  name: string;
  email: string;
  photoUrl: string;
  emailVerified: boolean;
  uid: string;
}

interface IStringMap {
  [s: string]: string;
}

type PicktusMessageLevelType =
  | PicktusMessageLevel.WARNING
  | PicktusMessageLevel.SUCCESS
  | PicktusMessageLevel.INFO
  | PicktusMessageLevel.ERROR;

type PicktusMessageDisplayType = PicktusMessageDisplay.SNACKBAR | PicktusMessageDisplay.NOTIFICATION;

interface IPicktusMessage {
  display: PicktusMessageDisplayType;
  level: PicktusMessageLevel;
  text: string;
  id: string;
}

interface IPicktusError extends IPicktusMessage {
  level: PicktusMessageLevel.ERROR;
}
