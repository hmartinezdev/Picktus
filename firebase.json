{
  "hosting": {
    "public": "src/public"
  },
  "functions": {
    "source": "src/functions"
  }
}
