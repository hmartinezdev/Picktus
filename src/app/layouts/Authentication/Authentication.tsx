import Logo from '@components/Logo';
import colors from '@constants/colors';
import { borderRadius, boxShadow } from '@constants/styles';
import React, { Component } from 'react';
import { Transition, TransitionGroup } from 'react-transition-group';

export interface IAuthenticationPropsType {
  status: string;
  children: JSX.Element;
}

class Authentication extends Component<IAuthenticationPropsType> {
  public render(): React.ReactElement<Authentication> {
    const { status, children } = this.props;

    return (
      <div className={`container container--${status}`}>
        <div className={`logo`}>
          <Logo />
        </div>
        <TransitionGroup className="authentication__content" appear={true}>
          {React.Children.map(children, (child: JSX.Element) => (
            <Transition key={child.key || ''} timeout={300} mountOnEnter={true} unmountOnExit={true} appear={true}>
              {(contentStatus) => <div className={`contentContainer contentContainer--${contentStatus}`}>{child}</div>}
            </Transition>
          ))}
        </TransitionGroup>
        <style jsx>
          {`
            .container {
              width: 100%;
              height: 100vh;
              display: flex;
              align-items: center;
              flex-direction: column;
              padding: 2rem 0;
              box-sizing: border-box;
              overflow: hidden;
              opacity: 1;
              transition: opacity 300ms ease-out;
            }

            .container--exiting,
            .container--exited {
              position: absolute;
              z-index: 0;
            }

            .contentContainer {
              transition: opacity 300ms ease-out;
            }

            .contentContainer--entering {
              animation: opacity 300ms ease-in;
            }

            .contentContainer--entered {
              opacity: 1;
            }

            .contentContainer--exiting,
            .contentContainer--exited {
              position: absolute;
              opacity: 0;
            }

            .logo {
              background-color: ${colors.primary};
              box-shadow: ${boxShadow};
              display: flex;
              position: relative;
              border-radius: ${borderRadius};
              transition: opacity 300ms ease-out;
            }

            :global(.authentication__content) {
              height: 100%;
              width: 100%;
              display: flex;
              flex-direction: column;
              justify-content: flex-start;
              align-items: center;
              margin-top: 2.5rem;
            }

            @keyframes opacity {
              0% {
                opacity: 0;
              }

              100% {
                opacity: 1;
              }
            }
          `}
        </style>
      </div>
    );
  }
}

export default Authentication;
