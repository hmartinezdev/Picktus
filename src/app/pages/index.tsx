import Home from '@containers/Home';
import React from 'react';
import { IPageProps } from './page.type';

export default (props: IPageProps): React.ReactElement<Home> => <Home {...props} />;
