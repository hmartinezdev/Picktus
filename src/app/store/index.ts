import { createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import reducers from './reducers';

export function initializeStore(initialState = {}) {
  return createStore(reducers, initialState, composeWithDevTools());
}
