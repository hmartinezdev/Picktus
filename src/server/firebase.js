/* tslint:disable: max-line-length */

module.exports = {
  auth_uri: 'https://accounts.google.com/o/oauth2/auth',
  client_email: 'firebase-adminsdk-vkv42@picktus-76101.iam.gserviceaccount.com',
  client_id: '106852938874480059583',
  client_x509_cert_url:
    'https://www.googleapis.com/robot/v1/metadata/x509/firebase-adminsdk-vkv42%40picktus-76101.iam.gserviceaccount.com',
  private_key:
    '-----BEGIN PRIVATE KEY-----\nMIIEvwIBADANBgkqhkiG9w0BAQEFAASCBKkwggSlAgEAAoIBAQDQ7AF17FSf7fac\n2tHFfrmMr/U+tmNhjjT9RUlEXcq+hExx+5aqPqP4RZvWxCqPrkRfxLUfRgmDnY4l\nR37JDEU8uA4jvvT/CRvGP65Q7oWVSFEvGinXAV+2JcYxZxuvypfb+f8vVz1ZJvjV\n9tCPNuPvr65tVnz6rR9G8bdoxLOmO4sWdUBt5HBR1P960hLVyMLF2TI0qV21Zfax\n6/J5JQh4cxnAVH9alFVYTwGGyaMK54J3nICrqC4tIL9orIjr4f0Ree49Koa2nK1S\nqrFL3nKlYuQ49MT5S8bsEkOZDI89x0cbE72z+9KSYX0wZisuqQS/mnwLgvo/V5KV\nyhmlaqp7AgMBAAECggEAGq0HwrbaV1FbG8wU3NLBfmU5Oziul9ZAjdY69K/FI25h\nFGy7YCF/bkP1EgcaKB5oPJIxTjzzGgsUC06FscYPUSFvLMFFJwtSsI2RBGRwSg9h\nzlHY5csXHXrVQ5mWv2tu2HejTbgl7Sw8uZU6LA8mK4STEQLqElfolnZmpMn45lCY\n3OejiCHlFiARiRV56Jr7DO5AlZzklusBXitDHUhIzCLBjjH0KdgqRrByRfSMNRTV\nawggV9ysqI5iOIAKdJXNTr6CoOBAAl7YJLiJ71y0KsNUOU06/Z6eCNAqcqS1yl/e\nCIzdsk3bnoZ8GG/TxjevYjf4/PfdXy+Q+QNFaWzwAQKBgQD0Di/ka1+5rVGNKeGU\noS6+EpFVXazrGFwxA6QQV/VXxl1aLIMqLJDmSr+TTHEJbYwV7WXIrcF+aewsgDkh\nUH2OJ7t36g4n1uEV0ZhzegymrGWwHlZFFVNp4Hmez2hbz+ibQTdPGBA+F8qNR8KK\neNkreEuShXE+ehqOrZZghN5s+wKBgQDbJZ/mim3K9KGPZ7sMNuZEF1cc2ArR8jiC\nV0F7E58bRJOdxiCOk/A3xr8gNEN8cMhB9Xxd7VmTs7GNUCb0b9IFC6V03/58Ka4N\n2aWeAZX4xtUYy8SMSP/Wsw4FIUjmHmZ9lP/5KIm/LLKDRr6qUKCv2+LiB8GhnLRx\nx0bceJVAgQKBgQDstFHY/hcy1BnzivpCGG5oZUPWAuvyfIb19u0J+PV482iZS/DP\nV7gV08upBovftAMZRU3h7Dxt6jfpFVfTquiLlpepfzXJiAx5B/cPkQFyKmTaAyJz\nEHL9jn0G/WzfF3AoHIGTUeh3AMYwbrgNa5ZeJedOLeSezvixB0WBSX0dPwKBgQCw\nTwbkPX6ziDjq3POZkhmTsot9QzsBSbS4+GZQFKsvDu8MQjgzPUZIOXfO+v1Igpw6\npLvX/XrR4AH6xP9yLPx03rfQ3d0qG4ZS0kXyZyg3LBCJkHaivqBGJWwkKHEz4kVm\nx5dg54Ja8iT0hNgkwhVlJqzUWJUp5p8t4rtZ6RF7AQKBgQC+7c90RFR7CEuJp8vQ\njjn8n+xGuaHXXEMEgarCGvLOs8UUTa5OrRDXY91KIyxh7hdUYLEMiR0XP4wIDF91\nflRfCE2SXsW33e/jd+oZkYCg5w1Oofj4XAyp0QoRRE60Z0MCPceq86tUVy0N5oeD\nyXokhCB/hpT3XtH2GXZiqhwQ3g==\n-----END PRIVATE KEY-----\n',
  private_key_id: '3140acfc3ecc44fc2114f9d17512fd82d49d1ac7',
  project_id: 'picktus-76101',

  token_uri: 'https://oauth2.googleapis.com/token',
  type: 'service_account',
};
