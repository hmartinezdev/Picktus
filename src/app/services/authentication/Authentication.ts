import { session } from '@constants/cookies';
import * as firebase from 'firebase/app';
import 'firebase/auth';
import fetch from 'isomorphic-unfetch';
import Cookies from 'js-cookie';
import {
  delegatedMethods,
  FirebaseErrorCodes,
  ISigninMethodsMap,
  ISigninOptions,
  SigninMethods,
} from './Authentication.type';
import AuthenticationError from './AuthenticationError';
import { FirebaseError } from './AuthenticationError/AuthenticationError.type';

class Authentication {
  public async createUser(email: string, password: string) {
    return firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .catch((error: FirebaseError) => {
        throw this.handleFirebaseError(error, 'createUser');
      });
  }

  /**
   * Function handling the the social network signin
   *
   * @param method - method of authentication
   */
  private async delegatedAuthentication(method: delegatedMethods) {
    const providers: ISigninMethodsMap = {
      [SigninMethods.FACEBOOK]: new firebase.auth.FacebookAuthProvider(),
      [SigninMethods.GOOGLE]: new firebase.auth.GoogleAuthProvider(),
      [SigninMethods.TWITTER]: new firebase.auth.TwitterAuthProvider(),
      [SigninMethods.CLASSIC]: undefined,
    };

    const provider = providers[method];

    // Use the navigator default language
    firebase.auth().useDeviceLanguage();

    if (method === SigninMethods.FACEBOOK) {
      provider.setCustomParameters({
        display: 'popup',
      });
    }

    return firebase
      .auth()
      .signInWithPopup(provider)
      .catch((error) => {
        throw this.handleFirebaseError(error, 'delegatedAuthentication');
      });
  }

  /**
   * Function handling the whole signin process
   *
   * @param method - method of authentication, can be either classic or via social networks
   * @param options - options needed to signin with a specific method
   */
  public async signin(method: SigninMethods, options: ISigninOptions = {}) {
    let userCredentials: firebase.auth.UserCredential;
    if (method === SigninMethods.CLASSIC) {
      if (!options.email || !options.password) {
        throw new AuthenticationError(
          "Authentication::signin You can't use classic signin method without password or email passed as option",
          'A password and an email is necessary to signin'
        );
      }

      userCredentials = await firebase
        .auth()
        .signInWithEmailAndPassword(options.email, options.password)
        .catch((error) => {
          throw this.handleFirebaseError(error, 'signin');
        });
    } else {
      userCredentials = await this.delegatedAuthentication(method);
    }

    if (!userCredentials || !userCredentials.user) {
      throw new AuthenticationError(
        `Authentication::signin No user retrieved from firebase`,
        'We have a problem with the authentication server'
      );
    }

    const user = userCredentials.user;

    // Get the user's ID token as it is needed to exchange for a session cookie.
    user.getIdToken().then((idToken) => {
      // Session login endpoint is queried and the session cookie is set.
      // CSRF protection should be taken into account.
      // ...

      return fetch('/sessionlogin', {
        body: JSON.stringify({ idToken }),
        headers: {
          'Content-Type': 'application/json',
        },
        method: 'POST',
      });
    });

    return user;
  }

  public async disconnect() {
    await firebase
      .auth()
      .signOut()
      .catch((error) => {
        throw this.handleFirebaseError(error, 'signout');
      });

    // Remove the cookie set in session
    Cookies.remove(session);

    // reloading page to ensure the user is disconnected
    if (window && window.location) {
      window.location.reload();
    }
  }

  /**
   * Handle the generation of formated error
   *
   * @param error - error dispatched by firebase
   * @param func - function from which the error has been thrown
   */
  public handleFirebaseError(error: FirebaseError, func: string) {
    switch (error.code) {
      case FirebaseErrorCodes.TOO_MANY_REQUEST:
        return new AuthenticationError(
          `Authentication::${func} too many requests`,
          'You tried too many time to signin, please try again later'
        );
      case FirebaseErrorCodes.NETWOR_REQUEST_FAILED:
        return new AuthenticationError(
          `Authentication::${func} network error`,
          'You are currently in a slow network area, please try again later'
        );
      case FirebaseErrorCodes.INVALID_API_KEY:
        return new AuthenticationError(
          `Authentication::${func} invalid API provided`,
          'A client error has occured, please contact the consumer service'
        );
      case FirebaseErrorCodes.INVALID_EMAIL:
        return new AuthenticationError(
          `Authentication::${func} invalid email provided by the user`,
          'The email you provided is not a valid email address'
        );
      case FirebaseErrorCodes.USER_NOT_FOUND:
        return new AuthenticationError(
          `Authentication::${func} email was not associated with an account`,
          'The email you provided is not associated with an existing account'
        );
      case FirebaseErrorCodes.WRONG_PASSWORD:
        return new AuthenticationError(
          `Authentication::${func} password provided did not correspond with the email`,
          'The password and email combination is not associated with an existing account'
        );
      case FirebaseErrorCodes.ACCOUNT_ALREADY_EXIST_WITH_DIFFERENT_CREDENTIAL:
      case FirebaseErrorCodes.EMAIL_ALREADY_USED:
        return new AuthenticationError(
          `Authentication::${func} an account already exists with the user email`,
          'An account is already associated with this email'
        );
      case FirebaseErrorCodes.POPUP_BLOCKED:
        return new AuthenticationError(
          `Authentication::${func} popin blocked by navigator`,
          'The authentication popin has been blocked by your navigator'
        );
      case FirebaseErrorCodes.WEAK_PASSWORD:
        return new AuthenticationError(
          `Authentication::${func} popin blocked by navigator`,
          'The authentication popin has been blocked by your navigator'
        );
      case FirebaseErrorCodes.CANCELLED_POPUP:
        return new AuthenticationError(
          `Authentication::${func} popin cancelled because a new one has been opened`,
          'The previous authentication popup has been closed because you opened a new one'
        );
      default:
        return new AuthenticationError(
          `Authentication::${func} Unknown error: ${error.code}::${error.message}`,
          'An unknown error has occurred'
        );
    }
  }
}

export default new Authentication();
