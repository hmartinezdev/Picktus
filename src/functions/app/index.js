import { server, app } from './server';
import * as functions from 'firebase-functions';

const dev = process.env.NODE_ENV !== 'production';

const nextApp = functions.https.onRequest(async (req, res) => {
  console.log('File: ' + req.originalUrl);
  console.log('dev:', dev);
  // log the page.js file or resource being requested

  await app.prepare();
  return server(req, res);
});

export { nextApp };
