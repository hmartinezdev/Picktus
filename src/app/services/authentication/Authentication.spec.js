import * as firebase from 'firebase/app';
import fetch from 'isomorphic-unfetch';
import { incorrectUserCredentials, validUserCredentials } from './Authentication.mock';
import Authentication, { SigninMethods } from './';

jest.mock('isomorphic-unfetch');

describe('Authentication service', () => {
  describe('createUser', () => {
    afterEach(() => {
      jest.restoreAllMocks();
    });

    test('it should call firebase to create the new user', async () => {
      jest
        .spyOn(firebase, 'auth')
        .mockImplementation(() => ({ createUserWithEmailAndPassword: () => new Promise((resolve) => resolve()) }));

      await Authentication.createUser('test', 'test');
      expect(firebase.auth).toHaveBeenCalledTimes(1);
    });

    test('it should throw an error if the user creation fails', async () => {
      jest.spyOn(firebase, 'auth').mockImplementation(() => ({
        createUserWithEmailAndPassword: () =>
          new Promise(() => {
            throw new Error('test');
          }),
      }));

      let error = null;

      try {
        await Authentication.createUser('test', 'test');
      } catch (e) {
        error = e;
      }

      expect(error instanceof Error).toBe(true);
    });
  });

  describe('signin', () => {
    afterEach(() => {
      jest.restoreAllMocks();
    });

    describe('with classic method', () => {
      test('it should throw an error if password or email is not defined in the options', async () => {
        let error;

        try {
          await Authentication.signin(SigninMethods.CLASSIC, { password: 'test' });
        } catch (e) {
          error = e;
        }

        expect(error instanceof Error).toBe(true);
        error = undefined;

        try {
          await Authentication.signin(SigninMethods.CLASSIC, {});
        } catch (e) {
          error = e;
        }

        expect(error instanceof Error).toBe(true);
        error = undefined;

        try {
          await Authentication.signin(SigninMethods.CLASSIC, { email: 'test' });
        } catch (e) {
          error = e;
        }

        expect(error instanceof Error).toBe(true);
      });

      test('it should throw an error if the user login fails', async () => {
        jest.spyOn(firebase, 'auth').mockImplementation(() => ({
          signInWithEmailAndPassword: () =>
            new Promise(() => {
              throw new Error('test');
            }),
        }));

        let error = null;

        try {
          await Authentication.signin(SigninMethods.CLASSIC, { email: 'email', password: 'password' });
        } catch (e) {
          error = e;
        }

        expect(error instanceof Error).toBe(true);
      });

      test('it should return the user if the authentication proceed correctly', async () => {
        const spySignin = jest.fn().mockImplementation(() => new Promise((resolve) => resolve(validUserCredentials)));
        jest.spyOn(firebase, 'auth').mockImplementation(() => ({ signInWithEmailAndPassword: spySignin }));

        const result = await Authentication.signin(SigninMethods.CLASSIC, { email: 'email', password: 'password' });
        fetch.mockResolvedValue(undefined);

        expect(spySignin).toHaveBeenCalledWith('email', 'password');
        expect(fetch).toHaveBeenCalledWith('/sessionlogin', {
          body: JSON.stringify({ idToken: 'idToken' }),
          headers: {
            'Content-Type': 'application/json',
          },
          method: 'POST',
        });
        expect(result).toEqual(validUserCredentials.user);
      });
    });

    describe('Delegated authentication', () => {
      let spyAuth;

      afterEach(() => {
        jest.restoreAllMocks();
      });

      beforeEach(() => {
        spyAuth = jest.spyOn(Authentication, 'delegatedAuthentication').mockImplementation(() => validUserCredentials);
        fetch.mockResolvedValue(undefined);
      });

      test('it should return the user if the facebook authentication proceed correctly', async () => {
        const result = await Authentication.signin(SigninMethods.FACEBOOK);

        expect(spyAuth).toHaveBeenCalledWith(SigninMethods.FACEBOOK);
        expect(fetch).toHaveBeenCalledWith('/sessionlogin', {
          body: JSON.stringify({ idToken: 'idToken' }),
          headers: {
            'Content-Type': 'application/json',
          },
          method: 'POST',
        });
        expect(result).toEqual(validUserCredentials.user);
      });
    });

    describe('General case', () => {
      afterEach(() => {
        jest.restoreAllMocks();
      });

      test('if no user is returned from auth, it should throw an error', async () => {
        const spySignin = jest.fn().mockImplementation(() => new Promise((resolve) => resolve()));
        jest.spyOn(firebase, 'auth').mockImplementation(() => ({ signInWithEmailAndPassword: spySignin }));

        try {
          await Authentication.signin(SigninMethods.CLASSIC, { email: 'email', password: 'password' });
        } catch (e) {
          expect(e instanceof Error).toBe(true);
        }
      });

      test('it should throw an error if the user credentials are incorrects', async () => {
        jest.spyOn(firebase, 'auth').mockImplementation(() => ({
          signInWithEmailAndPassword: () =>
            new Promise((resolve) => {
              resolve(incorrectUserCredentials);
            }),
        }));

        let error = null;

        try {
          await Authentication.signin(SigninMethods.CLASSIC, { email: 'email', password: 'password' });
        } catch (e) {
          error = e;
        }

        expect(error instanceof Error).toBe(true);
      });
    });
  });

  describe('delegatedAuthentication', () => {
    let spyUsedDevice = jest.fn();
    let spySigninWP = jest.fn().mockImplementation(() => new Promise((resolve) => resolve()));
    let spyCustomParm = jest.fn();
    let authProvider;
    beforeEach(() => {
      const auth = function() {
        return {
          signInWithPopup: spySigninWP,
          useDeviceLanguage: spyUsedDevice,
        };
      };

      authProvider = jest.spyOn(firebase, 'auth').mockImplementation(auth);
      authProvider.FacebookAuthProvider = class FacebookAuthProvider {
        setCustomParameters = spyCustomParm;
      };
      authProvider.GoogleAuthProvider = class GoogleAuthProvider {};
      authProvider.TwitterAuthProvider = class TwitterAuthProvider {};
    });

    afterEach(() => {
      jest.restoreAllMocks();
    });

    test('it should work with facebook authentication', async () => {
      await Authentication.delegatedAuthentication(SigninMethods.FACEBOOK);

      expect(firebase.auth).toHaveBeenCalledTimes(2);
      expect(spyUsedDevice).toHaveBeenCalled();
      expect(spyCustomParm).toHaveBeenCalledWith({
        display: 'popup',
      });
      expect(spySigninWP).toHaveBeenCalledWith(new authProvider.FacebookAuthProvider());
    });

    test('it should work with google authentication', async () => {
      await Authentication.delegatedAuthentication(SigninMethods.GOOGLE);

      expect(firebase.auth).toHaveBeenCalledTimes(2);
      expect(spyUsedDevice).toHaveBeenCalled();
      expect(spySigninWP).toHaveBeenCalledWith(new authProvider.GoogleAuthProvider());
    });

    test('it should work with twitter authentication', async () => {
      await Authentication.delegatedAuthentication(SigninMethods.TWITTER);

      expect(firebase.auth).toHaveBeenCalledTimes(2);
      expect(spyUsedDevice).toHaveBeenCalled();
      expect(spySigninWP).toHaveBeenCalledWith(new authProvider.TwitterAuthProvider());
    });

    test('it should throw an error if the user authentication fails', async () => {
      spySigninWP.mockImplementation(
        () =>
          new Promise(() => {
            throw new Error();
          })
      );

      let error = null;

      try {
        await Authentication.delegatedAuthentication(SigninMethods.FACEBOOK);
      } catch (e) {
        error = e;
      }

      expect(spyUsedDevice).toHaveBeenCalledWith();
      expect(spyCustomParm).toHaveBeenCalledWith({
        display: 'popup',
      });
      expect(error instanceof Error).toBe(true);
    });
  });
});
