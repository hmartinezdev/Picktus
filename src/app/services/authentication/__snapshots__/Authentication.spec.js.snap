// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication service handleFirebaseError error message to fit the error code 1`] = `[Error: Authentication::undefined Unknown error: undefined::undefined]`;

exports[`Authentication service handleFirebaseError error message to fit the error code 2`] = `[Error: Authentication::undefined too many requests]`;

exports[`Authentication service handleFirebaseError error message to fit the error code 3`] = `[Error: Authentication::undefined network error]`;

exports[`Authentication service handleFirebaseError error message to fit the error code 4`] = `[Error: Authentication::undefined invalid API provided]`;

exports[`Authentication service handleFirebaseError error message to fit the error code 5`] = `[Error: Authentication::undefined invalid email provided by the user]`;

exports[`Authentication service handleFirebaseError error message to fit the error code 6`] = `[Error: Authentication::undefined email was not associated with an account]`;

exports[`Authentication service handleFirebaseError error message to fit the error code 7`] = `[Error: Authentication::undefined password provided did not correspond with the email]`;

exports[`Authentication service handleFirebaseError error message to fit the error code 8`] = `[Error: Authentication::undefined an account already exists with the user email]`;

exports[`Authentication service handleFirebaseError error message to fit the error code 9`] = `[Error: Authentication::undefined popin blocked by navigator]`;

exports[`Authentication service handleFirebaseError error message to fit the error code 10`] = `[Error: Authentication::undefined popin blocked by navigator]`;

exports[`Authentication service handleFirebaseError error message to fit the error code 11`] = `[Error: Authentication::undefined popin cancelled because a new one has been opened]`;

exports[`Authentication service handleFirebaseError error message to fit the error code 12`] = `[Error: Authentication::undefined Unknown error: random::undefined]`;
